tools/build_node.js

```diff
@@ -41,10 +41,28 @@ async function buildCJSIndex(name, languages) {
 }
 
 async function buildNodeLanguage(language, options) {
+  const EMIT = `function emitWarning() {
+    if (!emitWarning.warned) {
+      emitWarning.warned = true;
+      process.emitWarning(
+        'Using file extension in specifier is deprecated, use "highlight.js/lib/languages/%%%%" instead'
+      );
+    }
+  }
```

Comment:
Alternatively you could put this in a separate module and `import` it in the `.js.js` files: that would reduce the size of the npm package, and only one warning would be printed out.

Reply:
I like the warning per file.  If someone has 5 broken requires, they should see 5 warnings. (ie, per call site)

The size difference isn't huge.